gammapy/datasets/flux_points.py

```diff
@@ -310,13 +382,84 @@ def flux_pred(self):
 
     def stat_array(self):
         """Fit statistic array."""
+        return self.available_stat_type[self.stat_type]()
+
+    def _stat_array_chi2(self):
+        """Chi2 statistics."""
         model = self.flux_pred()
         data = self.data.dnde.quantity
         try:
-            sigma = self.data.dnde_err
+            sigma = self.data.dnde_err.quantity
         except AttributeError:
-            sigma = (self.data.dnde_errn + self.data.dnde_errp) / 2
-        return ((data - model) / sigma.quantity).to_value("") ** 2
+            sigma = (self.data.dnde_errn + self.data.dnde_errp).quantity / 2
+        return ((data - model) / sigma).to_value("") ** 2
+
+    def _stat_array_profile(self):
+        """Estimate statitistic from interpolation of the likelihood profile."""
+        model = (self.flux_pred() / self.data.dnde_ref).to_value("")
+        stat = np.zeros(model.shape)
+        for idx, interp in enumerate(self._profile_interpolators):
+            stat[idx] = interp(model[idx])
+        return stat
+
+    def _get_valid_profile_interpolators(self):
+        value_scan = self.data.stat_scan.geom.axes["norm"].center
+        shape_axes = self.data.stat_scan.geom._shape[slice(3, None)]
+        interpolators = np.empty(shape_axes, dtype=object)
+        self._mask_valid = np.ones(self.data.dnde.data.shape, dtype=bool)
+        for idx in np.ndindex(shape_axes):
+            stat_scan = np.abs(
+                self.data.stat_scan.data[idx].squeeze()
```

Comment:
I think this is not working correctly. Running locally on my machine, I get this error 

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[14], line 1
----> 1 dataset_fp.stat_type =  "profile"
      2 dataset_fp.stat_type
      3 dataset_fp.models = model

File ~/Gammapy-dev/gammapy/gammapy/datasets/flux_points.py:187, in FluxPointsDataset.stat_type(self, stat_type)
    185     self.stat_kwargs.setdefault("interp_scale", "sqrt")
    186     self.stat_kwargs.setdefault("extrapolate", True)
--> 187     self._profile_interpolators = self._get_valid_profile_interpolators()
    188 self._stat_type = stat_type

File ~/Gammapy-dev/gammapy/gammapy/datasets/flux_points.py:412, in FluxPointsDataset._get_valid_profile_interpolators(self)
    409 self._mask_valid = np.ones(self.data.dnde.data.shape, dtype=bool)
    410 for idx in np.ndindex(shape_axes):
    411     stat_scan = np.abs(
--> 412         self.data.stat_scan.data[idx].squeeze()
    413         - self.data.stat.data[idx].squeeze()
    414     )
    415     self._mask_valid[idx] = np.all(np.isfinite(stat_scan))
    416     interpolators[idx] = interpolate_profile(
    417         value_scan,
    418         stat_scan,
    419         interp_scale=self.stat_kwargs["interp_scale"],
    420         extrapolate=self.stat_kwargs["extrapolate"],
    421     )

IndexError: index 3 is out of bounds for axis 1 with size 3
```

Possibly the same issue as #5155

Reply:
The shape was reversed solved with :
`shape_axes = self.stat_scan.geom._shape[slice(3, None)][::-1]`